File: api/src/main/java/api/platform/Enyity/Owner.java
package api.platform.Enyity;

import com.mengyunzhi.core.entity.YunzhiEntity;

import javax.persistence.*;

@Entity
public class Owner implements YunzhiEntity {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    /**
     * 对应用户
     */
    @OneToOne
    private User user;


    private String address;

    @OneToOne
    private GoodsType goodsType;


    public Long getId() {
        return id;
    }

    public String getAddress() {
        return address;
    }

    public User getUser() {
        return user;
    }

    public GoodsType getGoodsType() {
        return goodsType;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setGoodsType(GoodsType goodsType) {
        this.goodsType = goodsType;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: web/src/app/func/Owner.ts
/**
 * 司机实体
 */
import {GoodsType} from './GoodsType';
import {CarType} from './CarType';
import {User} from './User';

export class Owner {
  /** id */
  id: number;

  user = new User();

  goodsType: GoodsType;

  address: string;

  name: string;



  constructor(param?: {
    user?: User,
    id?: number,
    name?: string,
    goodsType?: GoodsType,
    address: string;
  }) {
    if (param) {
      this.user = param.user;
      this.id = param.id;
      this.name = param.name;
      this.goodsType = param.goodsType;
      this.address = param.address;
    }
  }
}
